shared/gh/js/views/gh.admin-batch-edit-date.js

```diff
@@ -169,44 +169,55 @@ define(['lodash', 'moment', 'gh.core', 'gh.api.config'], function(_, moment, gh,
             // For each selected week, add another day
             _.each($weeks, function(chk) {
                 _.defer(function() {
+
                     // Get the week number
                     var eventWeek = parseInt(chk.value, 10);
 
                     // Get the date by week and day
                     var dateByWeekAndDay = gh.utils.getDateByWeekAndDay(termName, eventWeek, dayOfTheWeek);
+                    if (!_.contains([2,3], dayOfTheWeek)) {
+                        dateByWeekAndDay = moment(dateByWeekAndDay).subtract({'weeks': 1});
+                    }
 
-                    var eventYear = dateByWeekAndDay.getFullYear();
-                    var eventMonth = dateByWeekAndDay.getMonth();
-                    var eventDay = dateByWeekAndDay.getDate();
-                    var eventStartHour = 13;
-                    var eventStartMinute = 0;
-                    var eventEndHour = 14;
-                    var eventEndMinute = 0;
-
-                    // Create the start date of the event
-                    var startDate = moment([eventYear, eventMonth, eventDay, eventStartHour, eventStartMinute, 0, 0]);
-                    // Create the end date of the event
-                    var endDate = moment([eventYear, eventMonth, eventDay, eventEndHour, eventEndMinute, 0, 0]);
-
-                    $(document).trigger('gh.batchedit.addevent', {
-                        'eventContainer': $('.gh-batch-edit-events-container[data-term="' + termName + '"]').find('tbody'),
-                        'eventObj': {
-                            'tempId': gh.utils.generateRandomString(), // The actual ID hasn't been generated yet
-                            'isNew': true, // Used in the template to know this one needs special handling
-                            'selected': true,
-                            'displayName': $('.gh-jeditable-series-title').text(),
-                            'end': gh.utils.convertUnixDatetoISODate(moment(endDate).toISOString()),
-                            'location': defaultLocation,
-                            'type': gh.config.events.default,
-                            'organisers': defaultOrganisers,
-                            'start': gh.utils.convertUnixDatetoISODate(moment(startDate).toISOString())
-                        },
-                        'startDate': startDate
-                    });
+                    // Only add a new event for the generated day if the day is within a term
+                    if (gh.utils.getTerm(gh.utils.convertISODatetoUnixDate(moment(dateByWeekAndDay).utc().format()), true)) {
```

Comment:
That's only when you add a new day. If you add a new week, OT event's will still be generated.